clang/lib/CConv/ProgramInfo.cpp

```diff
@@ -508,6 +508,12 @@ void ProgramInfo::addVariable(clang::DeclaratorDecl *D,
     // Function Decls have FVConstraints.
     FVConstraint *F = new FVConstraint(D, *this, *AstContext);
     F->setValidDecl();
+    // Handle a function returning a typedef'd type
+    // RETURN HERE
+    auto *Ret_PV = dyn_cast<PVConstraint>(F->getReturnVar());
+    auto Ret_Ty = FD->getReturnType();
+    checkTypedef(Ret_Ty.getTypePtr(), *AstContext, FD, Ret_PV);
```

Comment:
Is this where you are unifying the `typedef`'s `ConstraintVariable` with those of the various things that use it? Here you are doing it with the function return variable?

Reply:
Yes, because the function can return a typedef correct?

Reply:
Why can't a function return a typedef?

Reply:
A function can return a typedef. So we need to unify the return variable if it is a typedef?